src/datachain/lib/dc.py

```diff
@@ -1624,6 +1625,208 @@ def subtract(  # type: ignore[override]
+                not set, all columns will be used for comparison
+            right_compare: Optional column or list of columns
+                    for the `other` to compare to.
+            added (bool): Whether to return added rows in resulting chain.
+            deleted (bool): Whether to return deleted rows in resulting chain.
+            modified (bool): Whether to return modified rows in resulting chain.
+            unchanged (bool): Whether to return unchanged rows in resulting chain.
+            status_col (str): Name of the new column that is created in resulting chain
+                representing diff status.
+
+        Example:
+            ```py
+            diff = persons.diff(
+                new_persons,
+                on=["id"],
+                right_on=["other_id"],
+                compare=["name"],
+                added=True,
+                deleted=True,
+                modified=True,
+                unchanged=True,
+                status_col="diff"
+            )
+            ```
+        """
+        rname = "right_"
+
+        def _rprefix(c: str, rc: str) -> str:
+            """Returns prefix of right of two companion left - right columns
+            from merge. If companion columns have the same name then prefix will
+            be present in right column name, otherwise it won't.
+            """
+            return rname if c == rc else ""
+
+        def _to_list(obj: Union[str, Sequence[str]]) -> list[str]:
+            return [obj] if isinstance(obj, str) else list(obj)
+
+        if on is None:
+            raise ValueError("'on' must be specified")
+
+        on = _to_list(on)
+        if right_on:
+            right_on = _to_list(right_on)
+            if len(on) != len(right_on):
+                raise ValueError("'on' and 'right_on' must be have the same length")
+
+        if compare:
+            compare = _to_list(compare)
+
+        if right_compare:
+            if not compare:
+                raise ValueError(
+                    "'compare' must be defined if 'right_compare' is defined"
+                )
+
+            right_compare = _to_list(right_compare)
+            if len(compare) != len(right_compare):
+                raise ValueError(
+                    "'compare' and 'right_compare' must be have the same length"
+                )
+
+        num_statuses = sum(1 if s else 0 for s in [added, deleted, modified, unchanged])
+
+        if num_statuses > 1 and not status_col:
+            raise ValueError(
+                "Status column name is needed if more than one status is asked"
+            )
+
+        if num_statuses == 0:
+            raise ValueError(
+                "At least one of added, deleted, modified, unchanged flags must be set"
+            )
+
+        # we still need status column for internal implementation even if not
+        # needed in output
+        status_col = status_col or "sys__diff"
```

Comment:
`sys__diff` - it might be better to use some random column name like `diff_37a345`

Reply:
Done